### DISCUSSION TITLE
How to reconcile HelmChart when appVersion changed.

### DISCUSSION THREAD

**Discussion by belkevglaz** [fluxcd/flux2#4303] [Q&A]
Hi.

I wouldn't like to create ImagePolicy/ImageRepository for all my services, that are deployed on cluster. 
I'd like to use immutable HelmChart (chart version changing very rarely) that stored in HelmRepository and reconcile environment when chart **appVersion** changed. Is this possible ? 

Thanks.

**Comment by stefanprodan** [comment]
Put the helm charts in a container registry, switch the HelmRepository to type OCI, then you can use image automation on HelmReleases, see https://fluxcd.io/flux/cheatsheets/oci-artifacts/#helm-oci-automation

**Reply by belkevglaz** [reply]
@stefanprodan 
Thanks, but I know about ImageAutomation and that was my previous workable version of flux configuration. There was mention of reconciliation without ImageAutomation.
GitRepository has `reconcileStrategy: Revision` option to get new revision with immutable version of chart, but HelmRepository doesn't seem to have anything similar.

**Reply by stefanprodan** [reply]
You can set a semver range as the version of the chart, and Flux will automatically do an upgrade when you push a new chart version.

**Reply by belkevglaz** [reply]
I was taking about Chart.appVersion, but not Chart.version.

**Reply by stefanprodan** [reply]
Ahh, no that's not possible, app version is just some metadata that doesn't uniquely identifies a chart, we can only use chart version as that's the image tag in the registry.
